In [5]:
import warnings
warnings.filterwarnings('ignore')
import re
from tqdm import tqdm
import requests
from bs4 import BeautifulSoup
import os
import pandas as pd

In [6]:
input_url_df = pd.read_csv('input.csv')
input_url_df

URL_ID                                                URL
0      123.0  https://insights.blackcoffer.com/rise-of-telem...
1      321.0  https://insights.blackcoffer.com/rise-of-e-hea...
2     2345.0  https://insights.blackcoffer.com/rise-of-e-hea...
3     4321.0  https://insights.blackcoffer.com/rise-of-telem...
4      432.0  https://insights.blackcoffer.com/rise-of-telem...
..       ...                                                ...
109  50921.0  https://insights.blackcoffer.com/coronavirus-i...
110  51382.8  https://insights.blackcoffer.com/coronavirus-i...
111  51844.6  https://insights.blackcoffer.com/what-are-the-...
112  52306.4  https://insights.blackcoffer.com/marketing-dri...
113  52768.2  https://insights.blackcoffer.com/continued-dem...

[114 rows x 2 columns]

In [7]:
os.listdir()

['1691829231524.pdf',
 'input.csv',
 'output_data',
 'scraping.ipynb',
 'text_processing.ipynb',
 'venv']

In [8]:
for index, row in input_url_df.iterrows():
    url = row['URL']
    id = row['URL_ID']
    print(f'{id}: {url}')

123.0: https://insights.blackcoffer.com/rise-of-telemedicine-and-its-impact-on-livelihood-by-2040-3-2/
321.0: https://insights.blackcoffer.com/rise-of-e-health-and-its-impact-on-humans-by-the-year-2030/
2345.0: https://insights.blackcoffer.com/rise-of-e-health-and-its-imapct-on-humans-by-the-year-2030-2/
4321.0: https://insights.blackcoffer.com/rise-of-telemedicine-and-its-impact-on-livelihood-by-2040-2/
432.0: https://insights.blackcoffer.com/rise-of-telemedicine-and-its-impact-on-livelihood-by-2040-2-2/
2893.8: https://insights.blackcoffer.com/rise-of-chatbots-and-its-impact-on-customer-support-by-the-year-2040/
3355.6: https://insights.blackcoffer.com/rise-of-e-health-and-its-imapct-on-humans-by-the-year-2030/
3817.4: https://insights.blackcoffer.com/how-does-marketing-influence-businesses-and-consumers/
4279.2: https://insights.blackcoffer.com/how-advertisement-increase-your-market-value/
4741.0: https://insights.blackcoffer.com/negative-effects-of-marketing-on-society/
5202.8: htt

In [9]:
os.chdir('F:/My_Disk/python/Projects/NLP/')

In [10]:
os.chdir('F:/My_Disk/python/Projects/NLP/output_data/')
websites_having_different_structure = []

for index, row in input_url_df.iterrows():
    url = row['URL']
    url_id = row['URL_ID']

    page = requests.get(url)
    # if page.status_code == 200:
    soup = BeautifulSoup(page.content, "html.parser")
    try: 
        res = soup.find("h1", class_="entry-title").text.strip()
        res2 = soup.find("div", class_= "td-post-content tagdiv-type").text.strip()
        res3 = res + ". "+ res2

        with open(f'{url_id}.txt','w', encoding='utf-8') as text_file:
            text_file.write(res3)
    except:
        websites_having_different_structure.append(url_id)
        
    print(f'{index} url scraped')

0 url scraped
1 url scraped
2 url scraped
3 url scraped
4 url scraped
5 url scraped
6 url scraped
7 url scraped
8 url scraped
9 url scraped
10 url scraped
11 url scraped
12 url scraped
13 url scraped
14 url scraped
15 url scraped
16 url scraped
17 url scraped
18 url scraped
19 url scraped
20 url scraped
21 url scraped
22 url scraped
23 url scraped
24 url scraped
25 url scraped
26 url scraped
27 url scraped
28 url scraped
29 url scraped
30 url scraped
31 url scraped
32 url scraped
33 url scraped
34 url scraped
35 url scraped
36 url scraped
37 url scraped
38 url scraped
39 url scraped
40 url scraped
41 url scraped
42 url scraped
43 url scraped
44 url scraped
45 url scraped
46 url scraped
47 url scraped
48 url scraped
49 url scraped
50 url scraped
51 url scraped
52 url scraped
53 url scraped
54 url scraped
55 url scraped
56 url scraped
57 url scraped
58 url scraped
59 url scraped
60 url scraped
61 url scraped
62 url scraped
63 url scraped
64 url scraped
65 url scraped
66 url scraped
67 ur

In [11]:
websites_having_different_structure # list of those urls that have different website structure and will be scrapped differently

[2345.0,
 4279.2,
 8435.4,
 11668.0,
 14900.6,
 17671.4,
 33372.6,
 33834.4,
 37528.8,
 40761.4,
 41223.2,
 43070.4]

In [12]:
df = input_url_df[input_url_df['URL_ID'].isin(websites_having_different_structure)] # make a df of those url and their id's
print('Number of additional urls to be scraped: ',len(df))
for index, row in df.iterrows():
    url = row['URL']
    id = row['URL_ID']
    print(f'{index}: {id}: {url}')

Number of additional urls to be scraped:  12
2: 2345.0: https://insights.blackcoffer.com/rise-of-e-health-and-its-imapct-on-humans-by-the-year-2030-2/
8: 4279.2: https://insights.blackcoffer.com/how-advertisement-increase-your-market-value/
17: 8435.4: https://insights.blackcoffer.com/ai-in-healthcare-to-improve-patient-outcomes/
24: 11668.0: https://insights.blackcoffer.com/how-neural-networks-can-be-applied-in-various-areas-in-the-future/
31: 14900.6: https://insights.blackcoffer.com/future-of-work-how-ai-has-entered-the-workplace/
37: 17671.4: https://insights.blackcoffer.com/covid-19-environmental-impact-for-the-future/
71: 33372.6: https://insights.blackcoffer.com/human-rights-outlook/
72: 33834.4: https://insights.blackcoffer.com/how-voice-search-makes-your-business-a-successful-business/
80: 37528.8: https://insights.blackcoffer.com/estimating-the-impact-of-covid-19-on-the-world-of-work-3/
87: 40761.4: https://insights.blackcoffer.com/how-covid-19-is-impacting-payment-preference

In [13]:
os.chdir('F:/My_Disk/python/Projects/NLP/output_data/')
os.getcwd()

'F:\\My_Disk\\python\\Projects\\NLP\\output_data'

In [14]:
for index, row in df.iterrows():
    url = row['URL']
    url_id = row['URL_ID']
    try:
        page = requests.get(url)
        # if page.status_code == 200: 
        soup = BeautifulSoup(page.content, "html.parser")

        res = soup.find("h1",  class_="tdb-title-text").text.strip()
        res2 = soup.find_all("div", class_= "tdb-block-inner td-fix-index")[14].text.strip()
        res3 = res + ". "+ res2

        with open(f'{url_id}.txt','w', encoding='utf-8') as text_file:
            text_file.write(res3)

    except:
        print('404 error: page does not exits')
        
    print(f'{index} url scrapped')

2 url scrapped
8 url scrapped
17 url scrapped
404 error: page does not exits
24 url scrapped
31 url scrapped
404 error: page does not exits
37 url scrapped
71 url scrapped
72 url scrapped
80 url scrapped
87 url scrapped
88 url scrapped
92 url scrapped


### So, now we have 112 files created after scraping, corresponding to their url, 2 url's are borken.